packages/bot-web-ui/src/stores/save-modal-store.js

```diff
@@ -0,0 +1,66 @@
+import filesaver from 'file-saver';
+import { observable, action } from 'mobx';
+import { saveWorkspaceToRecent, save_types } from '@deriv/bot-skeleton';
+
+export default class SaveModalStore {
+    @observable is_save_modal_open = false;
+    @observable button_status = 0; // 0 - none, 1 - loading, 2 - completed
+
+    constructor(root_store) {
+        this.root_store = root_store;
+    }
+
+    @action.bound
+    toggleSaveModal() {
+        if (!this.is_save_modal_open) {
+            this.setButtonStatus(0);
+        }
+
+        this.is_save_modal_open = !this.is_save_modal_open;
+    }
+
+    @action.bound
+    async onConfirmSave({ is_local, save_as_collection }) {
+        this.setButtonStatus(1);
+
+        const { file_name } = this.root_store.toolbar;
+        const { saveFile } = this.root_store.google_drive;
+        const xml = Blockly.Xml.workspaceToDom(Blockly.derivWorkspace);
+
+        xml.setAttribute('is_dbot', 'true');
+        xml.setAttribute('collection', save_as_collection ? 'true' : 'false');
+
+        if (is_local) {
+            const data = Blockly.Xml.domToPrettyText(xml);
+            const blob = new Blob([data], { type: 'text/xml;charset=utf-8' });
+
+            filesaver.saveAs(blob, `${file_name}.xml`);
```

Comment:
We have this logic in `scratch/utils/index.js` (see `save()`), perhaps we could update it a bit so we can re-use it here?

Reply:
yeah make sense